### PR TITLE
Handle directories with vars as well

### PR DESCRIPTION
**Pull request by agx**
group_vars and host_vars can be arbitrary named files in directories
named like the host or group. Handle this case as well.

### CODE REVIEW

File: lib/ansible_spec/load_ansible.rb [hunk 1]
```diff
@@ -301,14 +301,20 @@ def self.get_hash_behaviour()
   # param: flag to extention
   #         true:  .yml extension is optional
   #         false: must have .yml extention
-  def self.load_vars_file(vars, path, check_no_ext = false)
+  def self.load_vars_file(vars, path, check_no_ext = false, allow_dir = false)
```
Comment (volanja) [inline lib/ansible_spec/load_ansible.rb]: `allow_dir` isn't use in this function.
Is this necessary?

File: lib/ansible_spec/load_ansible.rb [hunk 2]
```diff
@@ -301,14 +301,20 @@ def self.get_hash_behaviour()
...
     vars_file = path
     if check_no_ext && !File.exist?(vars_file)
       vars_file = path+".yml"
     end
     if File.exist?(vars_file)
-      yaml = YAML.load_file(vars_file)
-      vars = merge_variables(vars, yaml)
+      if File.directory?(vars_file)
```
Comment (volanja) [inline lib/ansible_spec/load_ansible.rb]: Could you add some test case?

**Comment by agx** [comment]
On Thu, Sep 29, 2016 at 08:55:59AM -0700, volanja wrote:
> volanja requested changes on this pull request.
> 
> Please check code.
> and Could you add some test case?
> 
> > @@ -301,14 +301,20 @@ def self.get_hash_behaviour()
>    # param: flag to extention
>    #         true:  .yml extension is optional
>    #         false: must have .yml extention
> -  def self.load_vars_file(vars, path, check_no_ext = false)
> +  def self.load_vars_file(vars, path, check_no_ext = false, allow_dir = false)
> 
> `allow_dir` isn't use in this function.
> Is this necessary?
> 
> >      vars_file = path
>      if check_no_ext && !File.exist?(vars_file)
>        vars_file = path+".yml"
>      end
>      if File.exist?(vars_file)
> -      yaml = YAML.load_file(vars_file)
> -      vars = merge_variables(vars, yaml)
> +      if File.directory?(vars_file)
> 
> Could you add some test case?

Can you suggest a file to add the test?

**Comment by volanja** [comment]
Please add in [spec/get_variable_spec.rb](https://github.com/volanja/ansible_spec/blob/master/spec/get_variable_spec.rb#L412)

**Comment by agx** [comment]
On Thu, Sep 29, 2016 at 09:07:54AM -0700, volanja wrote:
> Please add in [spec/get_variable_spec.rb](https://github.com/volanja/ansible_spec/blob/master/spec/get_variable_spec.rb#L412)

Thanks for the pointer. Test added!

**Comment by volanja** [comment]
Sorry for too late response.
I merged it.
Thank you for your contribution.

**Comment by agx** [comment]
Thanks and no worries. Any chance https://github.com/volanja/ansible_spec/pull/83 can be pulled in as well?

**Comment by volanja** [comment]
I merged #83 
it will release at v0.2.16

**Pull request merged**